### DISCUSSION TITLE
Support for Dynamic Properties on Records

### DISCUSSION THREAD

**Discussion by harshdoesdev** [fastn-stack/fastn#1147] [Ideas & RFCs]
Currently, we are facing a limitation where dynamic properties cannot be set on records, which is essential in various scenarios, particularly when working with Forms and APIs.

During the development of a form, I encountered this issue while creating a model for the KonfHub API (https://docs.konfhub.com/). The "registration_details" field contains an object that holds ticket details with a dynamic key, such as "ticket_id," which varies for each transaction.

Example of the current structure:

```javascript
"registration_details": {
    "3670": [
        {
            "name": "Bunny",
            "email_id": "bunny@example.com",
            "designation": "Dev",
            "organisation": "Uber",
            "linkedin_url": "linkedin.com/bunny"
        }
    ]
}
```

We need a way to define dynamic keys on an object, similar to how it is done in JavaScript:

```javascript
const ticket_id = 1234;

const reg_details = {
    [ticket_id]: {
        name: "Harsh Singh"
    }
}
```

This is what a possible implementation of this could look like in FTD.

```ftd
-- record ticket-details:
string name:
string email:
string phone:

-- record registration-details:
ticket-details [ticket-id]: 

-- ticket-details my-ticket-details:
name: Harsh Singh
email: something@example.com
phone: 1234567890

-- details event-registration-details:
dynamic ticket-id: 1234
ticket-details: $my-ticket-details
```

Here `[ticket-id]:` indicates that the key of this property is going to be dynamic, and in the `event-registration-details` we are passing the `ticket-id` with a keyword `dynamic` indicating that this is not a normal property/header, but value of a dynamic key that is going to be used by the record itself.

To overcome this limitation and enable more flexible data handling, we request the implementation of support for dynamic properties on records. This enhancement will significantly improve the usability and versatility of the platform, especially for scenarios involving dynamic data structures.

Thank you for considering this feature request.

**Comment by amitu** [comment]
@harshdoesdev do not open issues for feature requests, open a Discussion in [Ideas & RFCs](https://github.com/orgs/fastn-stack/discussions/categories/ideas-rfcs) category. 

Also I do not see any detail about what you want implemented. Share some FTD code you want working.

**Reply by harshdoesdev** [reply]
Thanks, I have converted the issue into a discussion now, and also added a sample code of what a possible implementation of this could look like in fastn.

**Comment by amitu** [comment]
@harshdoesdev what you are looking for is https://github.com/orgs/fastn-stack/discussions/507. Am closing this. Do re-open if you had something else in mind.
